### ISSUE TITLE
Make run_register optional?

### ISSUE THREAD

**Issue by aclark4life** [Jarn/jarn.mkrelease#3]
I'm trying to use `pypissh` to release a package. This works fine if I do:

```
$ python setup.py sdist upload
```

But fails if I try to use mkrelease, because (it appears) mkrelease unconditionally calls run_register, thus causing distutils to check for a username and password:

```running register
Traceback (most recent call last):
  File "<string>", line 41, in <module>
  File "setup.py", line 469, in <module>
    scripts=glob.glob("Scripts/pil*.py"),
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/distutils/core.py", line 152, in setup
    dist.run_commands()
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/distutils/dist.py", line 975, in run_commands
    self.run_command(cmd)
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/distutils/dist.py", line 995, in run_command
    cmd_obj.run()
  File "/Users/aclark/Developer/buildout.python/python-2.6/lib/python2.6/site-packages/distribute-0.6.24-py2.6.egg/setuptools/command/register.py", line 9, in run
    _register.run(self)
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/distutils/command/register.py", line 33, in run
    self._set_config()
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/distutils/command/register.py", line 75, in _set_config
    config = self._read_pypirc()
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/distutils/config.py", line 84, in _read_pypirc
    current['username'] = config.get(server, 'username')
  File "/Users/aclark/Developer/buildout.python/parts/opt/lib/python2.6/ConfigParser.py", line 541, in get
    raise NoOptionError(option, section)
ConfigParser.NoOptionError: No option 'username' in section: 'pypi'
register failed

```

Can the call to run_register be made optional? I don't see a way to do this in the options currently. Looking for something like:

    --skip-register

I can send a pull request if that would be more helpful
```

**Comment by petri** [comment]
+1 for this feature: I use pypiserver (https://pypi.python.org/pypi/pypiserver) which also barfs on 'register' command - while "python setup.py sdist upload" works fine.

**Comment by stefanholek** [comment]
FIxed in 3.10.
